qiskit/providers/models/pulsedefaults.py

```diff
@@ -94,44 +98,335 @@ class Command(BaseModel):
-        cmd_def (list[Command]): Backend command definition.
+    """Description of default settings for Pulse systems. These are operations or settings that
+    may be good starting points for the Pulse user. The user may modify these defaults through
+    the provided methods to build a reference to custom operations, which may in turn be used
+    for building Schedules or converting circuits to Schedules.
     """
 
-    def __init__(self, qubit_freq_est, meas_freq_est, buffer,
-                 pulse_library, cmd_def, **kwargs):
-        self.qubit_freq_est = qubit_freq_est
-        self.meas_freq_est = meas_freq_est
+    def __init__(self,
+                 qubit_freq_est: List[float],
+                 meas_freq_est: List[float],
+                 buffer: int,
+                 pulse_library: List[PulseLibraryItem],
+                 cmd_def: List[Command],
+                 **kwargs: Dict[str, Any]):
+        """
+        Validate and reformat transport layer inputs to initialize.
+
+        Args:
+            qubit_freq_est: Estimated qubit frequencies in GHz.
+            meas_freq_est: Estimated measurement cavity frequencies in GHz.
+            buffer: Default buffer time (in units of dt) between pulses.
+            pulse_library: Pulse name and sample definitions.
+            cmd_def: Operation name and definition in terms of Commands.
+            **kwargs: Other attributes for the super class.
+        """
+        super().__init__(**kwargs)
+
         self.buffer = buffer
+        self.qubit_freq_est = [freq * 1e9 for freq in qubit_freq_est]
+        self.meas_freq_est = [freq * 1e9 for freq in meas_freq_est]
         self.pulse_library = pulse_library
         self.cmd_def = cmd_def
+        self.ops_def = InstructionScheduleMap(cmd_def, pulse_library)
+
+    def __str__(self):
+        qubit_freqs = [freq / 1e9 for freq in self.qubit_freq_est]
+        meas_freqs = [freq / 1e9 for freq in self.meas_freq_est]
+        qfreq = "Qubit Frequencies [GHz]\n{freqs}".format(freqs=qubit_freqs)
+        mfreq = "Measurement Frequencies [GHz]\n{freqs} ".format(freqs=meas_freqs)
+        return ("<{name}({ops}{qfreq}\n{mfreq})>"
+                "".format(name=self.__class__.__name__, ops=str(self.ops_def),
+                          qfreq=qfreq, mfreq=mfreq))
+
+    def build_cmd_def(self) -> 'InstructionScheduleMap':
+        """
+        Return the InstructionScheduleMap built for this PulseDefaults instance.
 
-        super().__init__(**kwargs)
+        Returns:
+            InstructionScheduleMap: `InstructionScheduleMap` instance generated from defaults
+        """
+        warnings.warn("This method is deprecated. Returning a InstructionScheduleMap instead. "
+                      "This can be accessed simply through the `ops_def` attribute of this "
+                      "PulseDefaults instance.", DeprecationWarning)
+        return self.ops_def
+
+
+class InstructionScheduleMap():
```

Comment:
This name is up to discussion

Reply:
Unless the user wanted something from `backend.defaults()`, I don't see any reason for this class.
```
some_sched = Schedule('something that I built myself')
schedule = Schedule()
schedule += some_sched
```
vs
```
some_sched = Schedule('something that I built myself')
inst_map = backend.defaults().instruction_schedule_map  # Whatever we call this
qubits = [i for c.index in composable_schedule.channels]  # Might not need this line if it was saved when building some_sched
inst_map.add('name', qubits, some_sched)
schedule = Schedule()
schedule += inst_map.get('name', qubits)
```

Perhaps if `pulse` and `scheduler` have equal claim on it, it doesn't belong in either. Would you mind sharing with me what use cases you see for this class which wouldn't involve `PulseDefaults`? I'm probably missing something obvious

Reply:
If a user wished to create their own pulse calibration from scratch and schedule circuits using it, they would have to go looking in a completely weird backend focused location. The `scheduler` is built around this object, and we want to enable it to be extensible. We want to build a system that can be used from components, not one that requires all the pieces to be used.

Reply:
okay, where do you want it?

Reply:
Either top-level `pulse` or `scheduler`, I would lean towards `pulse` as that is where `cmd_def` is currently.

Reply:
+1 will do!